app/controllers/email_signins_controller.rb

```diff
@@ -2,23 +2,31 @@ class EmailSigninsController < ApplicationController
   def add
     supersede_existing_records
 
-    save_return = SaveReturn.find_by(service_slug: params[:service_slug],
-                                     encrypted_email: params[:encrypted_email])
+    save_return = SaveReturn.find_by(
+      service_slug: params[:service_slug],
+      encrypted_email: params[:encrypted_email]
+    )
 
     return render_email_missing_error unless save_return
 
-    magic_link = MagicLink.new(service_slug: params[:service_slug],
-                               encrypted_email: params[:encrypted_email],
-                               expires_at: expires_at)
+    magic_link = MagicLink.new(
+      service_slug: params[:service_slug],
+      encrypted_email: params[:encrypted_email],
+      expires_at: expires_at
+    )
 
     if magic_link.save
-      render json: { token: magic_link.id }, status: :created
+      return render json: { token: magic_link.id }, status: :created
+    else
+      return render_email_missing_error
```

Comment:
Added

Reply:
Are we catching the fact that either the service slug or the encrypted email might be missing from the payload?

Reply:
Nope. Not here, anyhoo

Let's look at that next week